numpy/ma/core.py

```diff
@@ -3184,13 +3180,15 @@ def __setitem__(self, indx, value):
 
         """
         # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
+        if (self._mask is not nomask) and \
+                (self._sharedmask or self._data.flags.owndata):
```

Comment:
By checking whether we own the data of the data part of the array, we can see if it is a view or not. If it is a view the mask should be a view in the future, so this is a good reason to check. This will only be worth checking if we don't already note that we share the mask.